docs/schemaform/form-config.md

```diff
@@ -0,0 +1,341 @@
+
+```js
+{
+  // We use this instead of the title property in the JSON Schema
+  'ui:title': '', 
+  
+  // We use this instead of the description property in the JSON Schema. This can be
+  // a string or a React component and would normally be used on object fields in the
+  // schema to provide description text or html before a block of fields
+  'ui:description': '' || DescriptionComponent,
+  
+  // Customize the field or widget you're using
+  'ui:field': '' || FieldComponent,
+  'ui:widget': '' || WidgetComponent,
+  
+  // This widget will be shown on the review page. Should always be used if you specify
+  // a custom widget component, but can be used with regular widgets as well. Currently
+  // only implemented for string fields
+  'ui:reviewWidget': WidgetComponent,
+
+  // Use this to provide a function to make a field conditionally required. First
+  // argument is the current form data and the second is the formContext object,
+  // which will contain the form data for other pages (tbd). You should avoid making
+  // a field required in the JSON schema and using `ui:required` on the same field.
+  'ui:required': function (pageData) {
+    return true || false;
+  },
+  
+  // This is an array of validation functions or objects that can be used to add validation
+  // that is not possible through JSON Schema. See below for the properties passed
+  // to the validation functions and how to use them.
+  'ui:validations': [
+    function (errors, fieldData, pageData, fieldSchema, errorMessages) {
+      errors.addError('My error');
+    },
+    {
+      validator: (errors, fieldData, pageData, fieldSchema, errorMessages, options) => {
+        errors.addError('My other error');
+      },
+      options: {}
+    }
+  ],
+  
+  // An object with field specific error messages. Structured by error name (from JSON
+  // Schema error types). This is passed to custom validations in `ui:validations` if
+  // you want to allow configurable error messages in a validator.
+  'ui:errorMessages': {
+    'pattern': 'Please provide a value in the right format'
+  },
+  'ui:options': {
+    
+    // This is a string of class names that will be added to the widget for the current
+    // field. Similar to the default `classNames` property, but will put the class names
+    // on the input/select/etc element itself, rather than a surrounding `div`.
+    widgetClassNames: '', 
+    
+    // For Array fields, this is a component that is shown when the item in the array is
+    // being shown read-only on a normal form page (i.e. not on the review page).
+    viewField: RowViewComponent, 
+    
+    // If you want a field to only be shown when another field is true, set this option
+    // to the property name. It will follow our ExpandingGroup pattern and expand underneath
+    // the field it is set to.
+    expandUnder: '', 
```

Comment:
For `expandUnder: 'someField'`, when `someField` is `true`, then this field will go under `someField`, correct? I'm not sure what the 'ExpandingGroup pattern` is exactly, and this seems to be the only reference in this doc.